pageserver/src/tenant/mgr.rs

```diff
@@ -1143,7 +1127,7 @@ impl TenantManager {
 
         match slot_guard.upsert(new_slot) {
             Err(TenantSlotUpsertError::InternalError(e)) => {
-                Err(UpsertLocationError::Other(anyhow::anyhow!(e)))
+                Err(UpsertLocationError::InternalError(anyhow::anyhow!(e)))
```

Comment:
The `anyhow!` captures the current stack trace.

That stack trace is not useless but the original InternalError might be more helpful during troubleshooting.

I.e., store an `Arc<anyhow::Error>` in the slot and return a clone of it here.

Reply:
The inner TenantSlotUpsertError contains a static string that describes each failure case -- I'd rather not carry a backtrace as well (the `anyhow` here is only because `InternalError` already requires it)